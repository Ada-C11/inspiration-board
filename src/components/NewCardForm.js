import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

class NewCardForm extends Component {
    constructor(props) {
        super(props);
        
        this.cleared = {
            text: "",
            emoji: ""
        };

        this.state = {...this.cleared}
        
        }

    addCard = (event) => {
        event.preventDefault();

        const card = this.state;

        this.props.addCardCallback(card)

        this.setState({...this.cleared});
    }

    onInputChange =(event) => {
        const updatedState = {};
        const field = event.target.name;
        const value = event.target.value;

        updatedState[field] = value;
        this.setState(updatedState);
    }    
    render() {
        const emoji = require("emoji-dictionary");

        const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]

        const emojiOption = EMOJI_LIST.map((symbol) => {
            return (
                <option value = {symbol}> 
                {emoji.getUnicode(symbol)}
                </option>
            )
        })
        return (
            <div className="new-card-form">
                <form className="new-card-form__form" onSubmit={this.addCard}>
                    <header className="new-card-form__header">
                        <h3>Add a Card</h3>
                    </header>
                    <label className="new-card-form__form-label">
                        Text:
                        <input
                        className="new-card-form__form-textarea"
                        name="text"
                        type="text"
                        value={this.state.name}
                        onChange={this.onInputChange}
                        >
                        </input>
                    </label>
                    <label className="new-card-form__form-label">
                        Emoji:
                        <select
                        className="new-card-form__form-select"
                        name="emoji"
                        value={this.state.emoji}
                        onChange={this.onInputChange}
                        >
                        {emojiOption}
                        </select>
                    </label>
                    <input className="new-card-form__form-button" 
                    type="submit"
                    value="Add a Card"
                    />
                </form>
            </div>
        )
    }
}


export default NewCardForm;