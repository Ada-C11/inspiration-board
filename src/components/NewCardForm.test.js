import React from 'react';
import NewCardForm from './NewCardForm.js';
import { shallow } from 'enzyme';

describe('NewCardForm', () => {
  test('that it matches an existing snapshot', () => {
    const wrapper = shallow(<NewCardForm addCardCallback={() => {}} />);

    expect(wrapper).toMatchSnapshot();
  });

  
});