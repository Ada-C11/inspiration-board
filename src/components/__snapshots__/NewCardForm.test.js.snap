// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCardForm that is matches an existing snapshot 1`] = `
<form
  className="new-card-form"
  onSubmit={[Function]}
>
  <h3>
     Add a Card
  </h3>
  <div>
    <label
      htmlFor="text"
    >
       Text: 
    </label>
    <input
      name="text"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="emoji"
    />
    <select
      name="emoji"
      onChange={[Function]}
      type="text"
      value=""
    >
      <option
        value=""
      >
        Add emojis to your message!
      </option>
      <option
        value=""
      >
        None
      </option>
      <option
        value="heart_eyes"
      >
        Heart Eyes
      </option>
      <option
        value="beer"
      >
        Beer
      </option>
      <option
        value="clap"
      >
        Clap
      </option>
      <option
        value="sparkling_heart"
      >
        Sparkling Heart
      </option>
      <option
        value="heart_eyes_cat"
      >
        Heart Eyes Cat
      </option>
      <option
        value="dog"
      >
        Dog
      </option>
    </select>
  </div>
  <div
    className=".new-card-form__form-button"
  >
    <input
      className="btn btn-success new-card-form--submit"
      name="submit"
      type="submit"
      value="Add a Card"
    />
  </div>
</form>
`;
